### DISCUSSION TITLE
Where are API settings stored (Docker)?

### DISCUSSION THREAD

**Discussion by ptz-nerf** [OvenMediaLabs/OvenMediaEngine#1140] [Help]
Hello!
When using REST API to change/create settings (create stream), where are these settings stored? In other words, how to organize persistent settings for Docker? The last time I tried after restarting all the changes were gone.

**Comment by Nepoh** [comment]
Inspecting a running OME docker container with `docker diff` reveals that only a `Server.id` file containing a single UUID and the server logs are created. Even after OME configuration has been changed via the API. So it looks like the server state is not persisted to the filesystem but only resides in memory.

```
C /opt
C /opt/ovenmediaengine
C /opt/ovenmediaengine/bin
C /opt/ovenmediaengine/bin/origin_conf
A /opt/ovenmediaengine/bin/origin_conf/Server.id
C /var
C /var/log
A /var/log/ovenmediaengine
A /var/log/ovenmediaengine/ovenmediaengine.log
```

**Comment by nilskretschmer** [comment]
As far as I know making changes to OME using the API those changes are not persistent.

We are using the API to add applications during runtime. If we restart the OME all applications are lost. I am not sure if this has changed in the latest versions?

**Comment by ptz-nerf** [comment]
Hello! Any changes?

**Comment by bchah** [comment]
@ptz-nerf API Storage is a feature of OvenMediaEngine Enterprise, which means that Airensoft likely will **not** bring this to the open-source version: https://ovenmediaengine-enterprise.gitbook.io/guide/enterprise-grade-features/advanced-management-tool/api-storage-or-0.17.0.0+

This leaves us 3 options:

1 - Buy an OvenMediaEngine Enterprise license from Airensoft.
2 - Create a PR for the same functionality which might convince them to allow it rather than manage divergent code of the same features.
3 - In your own application, store the API settings and make the required calls to restore them as part of the startup process.
